llama/runner/image.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ollama/ollama/llama"
+)
+
+// Locking: Mu must be held by the caller when generating image
+// embeddings
+
+type ImageContext struct {
+	Mu sync.Mutex
```

Comment:
What is the reason to leak this to other packages?

Reply:
Prior to this patch, there was an image cache that was grouped together with a larger input caching component but in reality was completely separate. With this change I introduced a new image component but wanted to share the lock between that and the cache, so the lock went in the image module but is used by other places.

I think a better approach is to move the image cache into the new image component and that component can do all of the locking itself without needing the caller to be aware of it and the mutex no longer needs to be exported. I made that change in an additional commit that I added to the series.